### ISSUE TITLE
ISSN 1522-5658 | Journal of Religion and Society | Importproblem

### ISSUE THREAD

**Issue by ixtheoOr** [ubtue/DatenProbleme#2160]
#### URL
http://moses.creighton.edu/JRS/toc/2024.html
  
#### Import-Translator
Mehrfachimport:
ubtue_Journal of Religion and Society.js

  
### Problembeschreibung
Es kommen keinerlei Metadaten mit. Es fehlen Titel, Verfasser, ISSN, Sprache, Jahr...

**Comment by mara-spiess** [comment]
@socheres Die Metadaten scheinen ursprünglich aus der DSpace Seite ausgelesen worden zu sein, auf der diese nun nicht mehr zu finden sind, bspw.: https://cdr.creighton.edu/items/b54b42da-babb-40a2-91a6-78c0b351f71d
Könnten Sie kurz prüfen ob sich tatsächlich etwas geändert hat und der Fehler daher stammt? Leider kenne ich die vorherige Version der Seiten nicht.

**Comment by mara-spiess** [comment]
Der neue Translator ubtue_Journal of Religion and Society_v2 sollte sowohl für die Zeitschrift als auch die Supplement Series (s. https://github.com/ubtue/DatenProbleme/issues/2165) funktionieren. Bitte Rückmeldung geben falls Änderungen ergänzt oder abgeändert werden sollen.

**Comment by ixtheoOr** [comment]
Der Translator funktioniert, allerdings werden beim aktuellen Heft 26.2004 drei von  acht Abstracts nicht mit importiert.

**Comment by socheres** [comment]
> @socheres Die Metadaten scheinen ursprünglich aus der DSpace Seite ausgelesen worden zu sein, auf der diese nun nicht mehr zu finden sind, bspw.: https://cdr.creighton.edu/items/b54b42da-babb-40a2-91a6-78c0b351f71d Könnten Sie kurz prüfen ob sich tatsächlich etwas geändert hat und der Fehler daher stammt? Leider kenne ich die vorherige Version der Seiten nicht.

Ja, die handle-URL aus dem getSearchResults wird als redirected URL aufgelöst und dort sind leider keine Metadaten mehr zu finden. Ich habe den Translator geändert, sodass die handle URL an die oai-pmh Schnittstelle übergeben wird. Dort sind strukturierte Metadaten zu finden (Auch zweiter Autor:innen werden anders als Journal of Religion and Society.js richtig erkannt).

**Comment by socheres** [comment]
@mara-spiess bitte den geänderten Translator testen und auf ready for testing setzen. Lokale Verzeichnisse wurden schon aktualisiert.

**Comment by socheres** [comment]
Ich habe mir die Ergebnisse vers. Formate angeschaut https://cdr.creighton.edu/server/oai/request?verb=ListMetadataFormats und im MARC-Format fehlt leider die Bandangabe: https://cdr.creighton.edu/oai/request?verb=GetRecord&metadataPrefix=marc&identifier=marc:cdr.creighton.edu:10504/154065

**Comment by jriedl** [comment]
Danke für die Erläuterung. Hatte u.a. gefragt, da DIM = DSpaceIntermediateMetadata und es Hinweise gibt, dass es nicht für das Harvesting genutzt werden sollte (vgl. z.B. https://wiki.archivematica.org/DSpace_exports)

**Comment by socheres** [comment]
Vielen Dank für den Hinweis. Die genannte Seite wird nicht mehr gepflegt: 'This page is no longer being maintained and may contain inaccurate information. Please see the Archivematica documentation for up-to-date information.' Leider finde ich in der aktuellen Archivematica-Dokumentation keine Informationen zum Format DIM: https://www.archivematica.org/docs/latest/
Von den 12 angebotenen Formaten enthalten nur DIM und XOAI Bandangaben. Zu XOAI fand ich diesen Hinweis:https://github.com/DSpace/xoai Daher ist DIM derzeit die beste Option.

**Comment by ixtheoOr** [comment]
An sich funktioniert der Import, aber:
1. Der Titel der Zeitschrift wird als Titelzusatz an den Titel des Aufsatzes gehängt
2. in Feld 2052 wird die URL und nicht nur der Handle eingetragen
3. 4950 wird doppelt eingetragen (H/R)

**Comment by ixtheoOr** [comment]
![Screenshot 2024-12-09 091403](https://github.com/user-attachments/assets/71b63893-df20-4ca8-b156-c42a43891e19)

**Comment by mara-spiess** [comment]
@socheres Leider kann ich nicht genau nachvollziehen warum der Export Translator 4950 doppelt generiert (mit jeweils unterschiedlichen internen Bemerkungen in $x).
Das Problem im Titel und das Format des Handles konnte ich problemlos beheben - erledigt sich mit dem korrekten Handle eventuell auch die Dopplung von 4950?

**Comment by mara-spiess** [comment]
@socheres auch hier besteht das selbe Problem bei den Anfragen an die OAI API, siehe https://github.com/ubtue/DatenProbleme/issues/2165#issuecomment-2545254755

**Comment by socheres** [comment]
Laut der aktuellen Code-Logik wird der Translator "ubtue_Journal of Religion and Society" auf der Seite Supplement gar nicht angeboten, weshalb er nicht die Ursache für das Problem sein kann (Test mit Run detect > 14:25:45 detectWeb did not match). Sollte der "ubtue_Journal of Religion and Society" jedoch auch für die Seite Supplement funktionieren, müsste der querySelector entsprechend angepasst werden. Also, bei der Funktion getSearchResults var text = doc.querySelectorAll('.title, .books a, p.chap'); ändern, dass er die spezifischen Selektoren der Supplement-Seite korrekt berücksichtigt. Bezüglich des Fehlers 4950 sollte der entsprechende Codeblock zu handle entfernt werden, um die Doppelausgabe zu beheben.

**Comment by ixtheoOr** [comment]
Ich habe in Zeder notiert, dass der Translator manchmal erst beim zweiten oder dirtten Versuch funktioniert!
